pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -6,21 +6,62 @@
 
 import argparse
 import cgi
+from datetime import datetime
 import hashlib
+import json
 import os
 from pathlib import Path
 import shutil
 import subprocess
 import sys
-from urllib import request
-from datetime import datetime
 from typing import Any, Dict
+from urllib import request
 
 
 from . import common
 from .io import parse_package_config
 
 
+class BashRunnerWithSharedEnvironment:
+    """Run multiple bash scripts with persisent environment.
+
+    Environment is stored to "env" member between runs. This can be updated
+    directly to adjust the environment, or read to get variables.
+    """
+
+    def __init__(self, env=None):
+        if env is None:
+            env = dict(os.environ)
```

Comment:
I'm not sure about this default. I think it might be better to explicitly pass the env variables needed. We have so many env variables set in various makefiles that it's a bit hard to follow. Normally none of those variables should be needed to build Python packages anyway I think?

Reply:
This keeps the logic in the file the same though. It was passing all of `os.environ` to all three stages (`build.script`, `compile` and `build.post`). I can move it so that it is passed as an argument?

Reply:
I figure because the whole environment was exposed before, if you want to change that it would be most appropriate to do so in a separate PR.

Reply:
OK, I though before we weren't passing the full env. Then never mind.